java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -356,57 +368,70 @@ private API getXPAPI(ExpressionTree et, VisitorState state) {
     return API.UNKNOWN;
   }
 
+  private Matcher<MethodInvocationTree> methodRecordMatcher(
+      MethodRecord methodRecord, VisitorState state) {
+
+    return allOf(
+        (methodRecord.isStatic() ? staticMethod().anyClass() : instanceMethod().anyClass())
+            .named(methodRecord.getMethodName()),
+        methodRecord
+            .getArgumentIdx()
+            .map(idx -> argument(idx, isArgumentMatchesFlagName(state)))
+            .orElseGet(Matchers::anything),
+        methodRecord
+            .getReceiverType()
+            .map(
+                type ->
+                    receiverOfInvocation((receiver, st) -> isSameType(type).matches(receiver, st)))
+            .orElseGet(Matchers::anything),
+        methodRecord.getReturnType().map(Matchers::isSameType).orElseGet(Matchers::anything));
+  }
+
+  private Matcher<ExpressionTree> enumConstructorArgsContainsFlagNameMatcher(
+      ImmutableCollection<PiranhaEnumRecord> enumRecords) {
+    Matcher<ExpressionTree> matchFlag =
+        (arg, stt) ->
+            arg.getKind() == Kind.STRING_LITERAL
+                && ((LiteralTree) arg).getValue().equals(xpFlagName);
+    return toType(
+        NewClassTree.class,
+        (nct, st) ->
+            enumRecords
+                .stream()
+                .anyMatch(
+                    enumRecord ->
```

Comment:
This is a bit over-nested, could we extract this lambda into a method we can use a method reference of here?

Reply:
If u see this code in your editor they are very small expressions. I don't kow why github makes them look so scary.

Reply:
You call here, I think :)